packages/server/src/calendar/calendar.controller.ts

```diff
@@ -51,32 +59,92 @@ export class CalendarController {
     }
   }
 
-  @Get(':cid')
-  async getAllEvents(@Param('cid') cid: number): Promise<CalendarModel[]> {
-    const events = await CalendarModel.find({
-      where: {
-        course: cid,
-      },
+  @Patch(':id')
+  async updateEvent(
+    @Param('id') id: string,
+    @Body() body: Partial<Calendar>,
+  ): Promise<CalendarModel> {
```

Comment:
Gonna need to pass in a courseId/cid and then use guards
```ts
@UseGuards(CourseRolesGuard)
  @Roles(Role.TA, Role.PROFESSOR)
```

ALSO: the `id` param MUST be changed to like calId or something, since CourseRolesGuard will see the `id` param and assume its a course id and then not good